overlay_error.py

```diff
@@ -0,0 +1,28 @@
+from overlay_new import OverlayNew, Overlay
+
+DEFAULT_TEST_BIKE = "v3"
+
+class TestOverlay(OverlayNew):
+    """
+    Overlay made for testing MQTT error publishing by using OverlayNew's MQTT client
+    
+    To exit the script, press CTRL+C.
+    """
+
+    def __init__(self, bike=DEFAULT_TEST_BIKE, bg=None):
+        super().__init__(bike, bg)
+    
+    def on_connect(self, client, userdata, flags, rc):
+        super().on_connect(client, userdata, flags, rc)
+
+        # Forcing an exception to occur
+        try: 
+            raise Exception("Test error")
+        except:
+            self.backend.send_camera_error(True) # Set to true to block the broker until error is published
```

Comment:
`paho-mqtt` is not in a fully connected state until the `on_connect` callback has returned. It also cannot publish until it is fully connected. However, the `on_connect` callback is unable to return until the message is published. So, we won't publish until we're connected, we won't connect until we've published, so nothing ever happens. If you change `True` to `False` here then everything works fine :)

Reply:
Hm, for some reason messages are able to be published on your computer, whereas mine doesn't seem to publish. Do you have time to troubleshoot this sometime before next software meeting? @hallgchris

Reply:
Sure thing! Will organise via slack